src/lematerial_forgebench/metrics/conditional_generation.py

```diff
@@ -0,0 +1,586 @@
+
+    DISTANCE_METRICS = {
+        "absolute": lambda x, target: abs(x - target),
+        "relative": lambda x, target: abs(x - target)
+        / (abs(target) if target != 0 else 1.0),
+        "squared": lambda x, target: (x - target) ** 2,
+        "identity": lambda x, target: x,
+    }
+
+    def __init__(
+        self,
+        target_value: float,
+        tolerance: float | None = None,
+        top_k: int | None = None,
+        distance_metric: str = "absolute",
+        name: str | None = None,
+        description: str | None = None,
+        n_jobs: int = 1,
+        lower_is_better: bool | None = None,  # Added parameter
+    ):
+        # If tolerance is None, the metric uses distance so lower is better
+        if lower_is_better is None:
+            lower_is_better = tolerance is None
+
+        if distance_metric not in self.DISTANCE_METRICS:
+            raise ValueError(
+                f"Invalid distance metric '{distance_metric}'. "
+                f"Must be one of: {list(self.DISTANCE_METRICS.keys())}"
+            )
+
+        super().__init__(
+            name=name or f"ContinuousTarget_{str(target_value)}",
+            description=description
+            or f"Measures {distance_metric} distance to target value {str(target_value)}",
+            lower_is_better=lower_is_better,
+            n_jobs=n_jobs,
+        )
+        self.config = ContinuousTargetConfig(
+            name=self.config.name,
+            description=self.config.description,
+            lower_is_better=self.config.lower_is_better,
+            n_jobs=self.config.n_jobs,
+            target_value=target_value,
+            tolerance=tolerance,
+            top_k=top_k,
+            distance_metric=distance_metric,
+        )
+
+    def _get_compute_attributes(self) -> dict[str, Any]:
+        """Get the attributes for the compute_structure method."""
+        return {
+            "target_value": self.config.target_value,
+            "distance_metric": self.config.distance_metric,
+            "cls": self.__class__,
+        }
+
+    @staticmethod
+    def value_extractor(structure: Structure) -> float:
+        """Extract the relevant continuous value from a structure.
+
+        This method must be implemented by subclasses.
+        """
+        raise NotImplementedError("Subclasses must implement value_extractor")
+
+    @staticmethod
+    def compute_structure(
+        structure: Structure, target_value: float, distance_metric: str, cls, **kwargs
+    ) -> float:
+        """Computes the distance from the target value using the specified metric."""
+        value = cls.value_extractor(structure, **kwargs)
+        return cls.DISTANCE_METRICS[distance_metric](value, target_value)
+
+    def aggregate_results(self, values: list[float]) -> Dict[str, Any]:
+        """Aggregate results into final metric values."""
+        valid_values = [v for v in values if not np.isnan(v)]
```

Comment:
Same, working with arrays would be faster.

Even such a simple manipulation is 10x faster with arrays:

```python
In [1]: import numpy as np

In [2]: data = np.random.randint(0, 100, (10000,)).astype(float)

In [3]: data[data<5] = float("nan")

In [4]: values = data.tolist()

In [5]: %timeit [v for v in values if not np.isnan(v)]
3.73 ms ± 67.9 µs per loop (mean ± std. dev. of 7 runs, 100 loops each)

In [6]: %timeit (w:= np.array(values))[~np.isnan(w)]
267 µs ± 1.33 µs per loop (mean ± std. dev. of 7 runs, 1,000 loops each)
```